pwd-checker.js

```diff
@@ -15,3 +15,44 @@ Focus: Use conditions to validate different password criteria and loops to analy
 */
 
 const prompt = require('prompt-sync')();
+console.log(`Welcome to the Password Strength Checker.`)
+
+let length = 0;
+let password;
+while (length === 0) {
+    password = prompt('Enter your password:');
+    length = password.length;
+}
+
+if (length < 8) {
+    console.log(`Password too short. Minimum length is 8.`)
+} else {
+     let strength = 1;
+    let hasNumbers = /\d/.test(password);
+    if (hasNumbers) {
+        strength += 3;
+    }
+    let hasUpper = /[A-Z]/.test(password);
+    if (hasUpper) {
+        strength += 3;
+    }
+    let hasSpecial = /[^a-zA-Z0-9]/.test(password);
+    if (hasSpecial) {
+        strength += 3;
```

Comment:
Skus to nespravit s Test ale skusime to spravit cez forEach a rozobrat si string bez regexu ale riesenie mas dobre

Reply:
pwd-checker je opravený pomocí forEach bez použití regexu.